lib/grpc/server.ex

```diff
@@ -203,13 +206,50 @@ defmodule GRPC.Server do
       next.(req, stream)
     rescue
       e in GRPC.RPCError ->
+        duration = System.monotonic_time() - t0
+
+        :ok =
+          GRPC.Telemetry.server_rpc_exception(
```

Comment:
Isn't `:telemetry.span` suitable for this case?

Reply:
`:telemetry.span` doesn't treat `rescue` separately

Reply:
Actually, I tested locally that `raise` can be `catch`-ed as an `:error`, so it seems that it's indeed equivalent, though I have to check if the only metadata that changes is what span handles :) I'll keep this unresolved til I confirm this.

Reply:
There were some changes needed anyway, but I went with span because it decouples a bit of the code and leaves basically the formatting and error-case handling to us.